### DISCUSSION TITLE
Questions about compilation process

### DISCUSSION THREAD

**Discussion by coldbootinitheart** [evilsocket/opensnitch#492] [Development]
My goal was to compile opensnitch on Arch Linux without having to install dependencies from pip.
All dependencies except `grpcio-tools`, I installed from pacman repos.
I blanked out the requirements.txt file for the ui and `make` and `sudo make install` without any problems.
Does the absence of `grpcio-tools` hamper any functionality? Opensnitch seems to be working fine for me but is there some functionality that does not work without `grpcio-tools`?

**Comment by gustavo-iniguez-goya** [comment]
Hi @coldbootinitheart ,

https://grpc.io/docs/languages/python/quickstart/#grpc-tools
> Python’s gRPC tools include the protocol buffer compiler protoc and the special plugin for generating server and client code from .proto service definitions.

We need `protoc` to generate the code that communicates client and server:
https://github.com/evilsocket/opensnitch/blob/master/proto/Makefile

The python command of that Makefile generates de protobuffer, take a look at it at `/usr/lib/python3/dist-packages/opensnitch/{ui_pb2.py,ui_pb3_grpc.py} Without these files daemon and GUI can't communicate.

Probably you had it already installed, or python3-protobuf and python3-grpcio installed separately.

**Comment by coldbootinitheart** [comment]
ah makes complete sense! I do indeed have python3-protobuf and python3-grpcio installed separately.
On another note while I do trust you guys as developers of opensnitch, you also take some go dependencies from 3rd parties.
Excuse my stupid question but can they be trusted? Obviously we need them for opensnitch to work in the first place but do you audit those dependencies?

**Reply by themighty1** [reply]
That's a good point. I think we should audit them at least once to make sure they don't phone home or do any obvious evil stuff  and then pin a hash of the package with the go.sum file.

**Reply by coldbootinitheart** [reply]
should I open an issue for auditing dependencies?

**Reply by themighty1** [reply]
it depends if @gustavo-iniguez-goya agrees that we need to pin down the go package hash. Maybe there is a drawback why this is not desirable, Im not sure. Then when the hash is chosen, we can create a discussion thread where people (who have coding experience) can make notes about having audited such-and-such package version.

**Reply by themighty1** [reply]
I, for one, think it would be a fun excercise I could allocate a few minutes per day towards.

**Reply by gustavo-iniguez-goya** [reply]
It's ok for me! Is there any difference on pinning down the package hash or use a tag? Also, dependding of what feature we use we'll depend on a hash or another. 

For example the Reject action will depend on netlink's https://github.com/evilsocket/opensnitch/commit/1b8ec1644149191c0b561ca79d0898c01003ec70#diff-fb123b2d60ee4889a2428abfc5aff499150e74a5d11da01ddfa842a582991497R13
